### PR TITLE
[is-expression] add types

### PR DESCRIPTION
**Pull request by tristan-f-r**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/is-expression/is-expression-tests.ts
1. types/is-expression/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9c0ecf1b6caa9b8cb0c7b3ed889c9f99c1b9c651
</p>

File: types/is-expression/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "private": true,
+    "name": "@types/is-expression",
+    "version": "4.0.9999",
+    "projects": [
+        "https://github.com/pugjs/is-expression#readme"
+    ],
+    "dependencies": {
+        "acorn": "7.1"
```
Comment (peterblazejewicz) [inline types/is-expression/package.json:9]: it's not the same as in upstream (only 7.1 pass the check).
https://github.com/pugjs/is-expression/blob/51045ecb52ad291d9df093c332debf28f64b726e/package.json#L13C16-L13C19

**Review by jakebailey — CHANGES_REQUESTED** [review]
After #67332, this PR will need to be updated to include an `.npmignore`. See https://github.com/DefinitelyTyped/DefinitelyTyped#npmignore

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@LeoDog896 thanks!

**Comment by LeoDog896** [comment]
Ready to merge

**Pull request merged**